tokio branch does not compile after tokio_service mutability change

i am gonna try to write a PR to fix this! here's the error for posterity.

```
error[E0053]: method `call` has an incompatible type for trait
  --> src/client/connect.rs:81:5
   |
81 |     fn call(&mut self, url: Url) -> Self::Future {
   |     ^ types differ in mutability
   |
   = note: expected type `fn(&client::connect::HttpConnector, url::Url) -> client::connect::HttpConnecting`
   = note:    found type `fn(&mut client::connect::HttpConnector, url::Url) -> client::connect::HttpConnecting`

error[E0053]: method `call` has an incompatible type for trait
   --> src/client/connect.rs:133:5
    |
133 |     fn call(&mut self, url: Url) -> Self::Future {
    |     ^ types differ in mutability
    |
    = note: expected type `fn(&client::connect::HttpsConnector, url::Url) -> client::connect::HttpsConnecting`
    = note:    found type `fn(&mut client::connect::HttpsConnector, url::Url) -> client::connect::HttpsConnecting`

error[E0053]: method `call` has an incompatible type for trait
   --> src/client/mod.rs:121:5
    |
121 |     fn call(&mut self, req: Request) -> Self::Future {
    |     ^ types differ in mutability
    |
    = note: expected type `fn(&client::Client<C>, client::request::Request) -> client::FutureResponse`
    = note:    found type `fn(&mut client::Client<C>, client::request::Request) -> client::FutureResponse`

error[E0053]: method `call` has an incompatible type for trait
   --> src/server/mod.rs:270:5
    |
270 |       fn call(&mut self, message: Self::Request) -> Self::Future {
    |  _____^ starting here...
271 | |         let (head, body) = match message {
272 | |             Message::WithoutBody(head) => (head, http::Body::empty()),
273 | |             Message::WithBody(head, body) => (head, body.into()),
274 | |         };
275 | |         let req = request::new(self.remote_addr, head, body);
276 | |         self.inner.call(req).map(map_response_to_message)
277 | |     }
    | |_____^ ...ending here: types differ in mutability
    |
    = note: expected type `fn(&server::HttpService<T>, tokio_proto::streaming::Message<http::MessageHead<http::RequestLine>, tokio_proto::streaming::Body<http::chunk::Chunk, error::Error>>) -> futures::Map<<T as tokio_service::Service>::Future, fn(server::response::Response) -> tokio_proto::streaming::Message<http::MessageHead<status::StatusCode>, tokio_proto::streaming::Body<http::chunk::Chunk, error::Error>>>`
    = note:    found type `fn(&mut server::HttpService<T>, tokio_proto::streaming::Message<http::MessageHead<http::RequestLine>, tokio_proto::streaming::Body<http::chunk::Chunk, error::Error>>) -> futures::Map<<T as tokio_service::Service>::Future, fn(server::response::Response) -> tokio_proto::streaming::Message<http::MessageHead<status::StatusCode>, tokio_proto::streaming::Body<http::chunk::Chunk, error::Error>>>`

error: aborting due to 4 previous errors

error: Could not compile `hyper`.

Caused by:
  process didn't exit successfully: `rustc --crate-name hyper src/lib.rs --crate-type lib -g --cfg feature="default" -C metadata=d4a40b29437ca96f -C extra-filename=-d4a40b29437ca96f --out-dir /Users/ag_dubs/rust/hyper/target/debug/deps --emit=dep-info,link -L dependency=/Users/ag_dubs/rust/hyper/target/debug/deps --extern tokio_tls=/Users/ag_dubs/rust/hyper/target/debug/deps/libtokio_tls-f3e5968fda14b2c3.rlib --extern httparse=/Users/ag_dubs/rust/hyper/target/debug/deps/libhttparse-996fc9116d9dcf83.rlib --extern url=/Users/ag_dubs/rust/hyper/target/debug/deps/liburl-8aa0752fa6071745.rlib --extern vecio=/Users/ag_dubs/rust/hyper/target/debug/deps/libvecio-b90c1aba84926426.rlib --extern mime=/Users/ag_dubs/rust/hyper/target/debug/deps/libmime-c64500266507952e.rlib --extern tokio_core=/Users/ag_dubs/rust/hyper/target/debug/deps/libtokio_core-a0dfb4e4436a27df.rlib --extern unicase=/Users/ag_dubs/rust/hyper/target/debug/deps/libunicase-ceb43341bcf99e15.rlib --extern native_tls=/Users/ag_dubs/rust/hyper/target/debug/deps/libnative_tls-2758c2463fafa3da.rlib --extern time=/Users/ag_dubs/rust/hyper/target/debug/deps/libtime-02018c58facc89ee.rlib --extern tokio_service=/Users/ag_dubs/rust/hyper/target/debug/deps/libtokio_service-94bffe2000aad1d6.rlib --extern futures_cpupool=/Users/ag_dubs/rust/hyper/target/debug/deps/libfutures_cpupool-0057d211711ed4da.rlib --extern rustc_serialize=/Users/ag_dubs/rust/hyper/target/debug/deps/librustc_serialize-03b62901d97343ea.rlib --extern language_tags=/Users/ag_dubs/rust/hyper/target/debug/deps/liblanguage_tags-69c1ab0d09b59b1d.rlib --extern log=/Users/ag_dubs/rust/hyper/target/debug/deps/liblog-1ce22d3a92f37841.rlib --extern cookie=/Users/ag_dubs/rust/hyper/target/debug/deps/libcookie-a1e4ecdfdab40c7f.rlib --extern tokio_proto=/Users/ag_dubs/rust/hyper/target/debug/deps/libtokio_proto-872bc602f52ffd09.rlib --extern relay=/Users/ag_dubs/rust/hyper/target/debug/deps/librelay-3c2ac943d6a43c65.rlib --extern mio=/Users/ag_dubs/rust/hyper/target/debug/deps/libmio-9b8eb2d3b7c024fa.rlib --extern futures=/Users/ag_dubs/rust/hyper/target/debug/deps/libfutures-986bba8c0d767260.rlib` (exit code: 101)
```

--- a/examples/hello.rs
+++ b/examples/hello.rs
@@ -17,7 +17,7 @@ impl Service for Hello {
     type Response = Response;
     type Error = hyper::Error;
     type Future = ::futures::Finished<Response, hyper::Error>;
-    fn call(&mut self, _req: Request) -> Self::Future {
+    fn call(&self, _req: Request) -> Self::Future {
         ::futures::finished(
             Response::new()
                 .with_header(ContentLength(PHRASE.len() as u64))

--- a/examples/server.rs
+++ b/examples/server.rs
@@ -21,7 +21,7 @@ impl Service for Echo {
     type Error = hyper::Error;
     type Future = ::futures::Finished<Response, hyper::Error>;
 
-    fn call(&mut self, req: Request) -> Self::Future {
+    fn call(&self, req: Request) -> Self::Future {
         ::futures::finished(match (req.method(), req.path()) {
             (&Get, Some("/")) | (&Get, Some("/echo")) => {
                 Response::new()

--- a/src/client/connect.rs
+++ b/src/client/connect.rs
@@ -29,7 +29,7 @@ pub trait Connect: Service<Request=Url, Error=io::Error> + 'static {
     /// A Future that will resolve to the connected Stream.
     type Future: Future<Item=Self::Output, Error=io::Error> + 'static;
     /// Connect to a remote address.
-    fn connect(&mut self, Url) -> <Self as Connect>::Future;
+    fn connect(&self, Url) -> <Self as Connect>::Future;
 }
 
 impl<T> Connect for T
@@ -40,7 +40,7 @@ where T: Service<Request=Url, Error=io::Error> + 'static,
     type Output = T::Response;
     type Future = T::Future;
 
-    fn connect(&mut self, url: Url) -> <Self as Connect>::Future {
+    fn connect(&self, url: Url) -> <Self as Connect>::Future {
         self.call(url)
     }
 }
@@ -78,7 +78,7 @@ impl Service for HttpConnector {
     type Error = io::Error;
     type Future = HttpConnecting;
 
-    fn call(&mut self, url: Url) -> Self::Future {
+    fn call(&self, url: Url) -> Self::Future {
         debug!("Http::connect({:?})", url);
         let host = match url.host_str() {
             Some(s) => s,
@@ -130,7 +130,7 @@ impl Service for HttpsConnector {
     type Error = io::Error;
     type Future = HttpsConnecting;
 
-    fn call(&mut self, url: Url) -> Self::Future {
+    fn call(&self, url: Url) -> Self::Future {
         debug!("Https::connect({:?})", url);
         let is_https = url.scheme() == "https";
         let host = match url.host_str() {
@@ -344,7 +344,7 @@ mod tests {
     fn test_non_http_url() {
         let mut core = Core::new().unwrap();
         let url = Url::parse("file:///home/sean/foo.txt").unwrap();
-        let mut connector = HttpConnector::new(1, &core.handle());
+        let connector = HttpConnector::new(1, &core.handle());
 
         assert_eq!(core.run(connector.connect(url)).unwrap_err().kind(), io::ErrorKind::InvalidInput);
     }

--- a/src/client/mod.rs
+++ b/src/client/mod.rs
@@ -118,7 +118,7 @@ impl<C: Connect> Service for Client<C> {
     type Error = ::Error;
     type Future = FutureResponse;
 
-    fn call(&mut self, req: Request) -> Self::Future {
+    fn call(&self, req: Request) -> Self::Future {
         let url = match req.uri() {
             &::RequestUri::AbsoluteUri(ref u) => u.clone(),
             _ => unimplemented!("RequestUri::*")
@@ -164,7 +164,7 @@ impl<C: Connect> Service for Client<C> {
                 // never had a pooled stream at all
                 e.into()
             });
-        let req = race.and_then(move |mut client| {
+        let req = race.and_then(move |client| {
             let msg = match body {
                 Some(body) => {
                     Message::WithBody(head, body.into())

--- a/src/client/pool.rs
+++ b/src/client/pool.rs
@@ -34,7 +34,7 @@ impl<T: Clone> Pool<T> {
         }
     }
 
-    pub fn checkout(&mut self, key: &str) -> Checkout<T> {
+    pub fn checkout(&self, key: &str) -> Checkout<T> {
         Checkout {
             key: Rc::new(key.to_owned()),
             pool: self.clone(),
@@ -284,7 +284,7 @@ mod tests {
 
     #[test]
     fn test_pool_checkout_smoke() {
-        let mut pool = Pool::new(true, Some(Duration::from_secs(5)));
+        let pool = Pool::new(true, Some(Duration::from_secs(5)));
         let key = Rc::new("foo".to_string());
         let mut pooled = pool.pooled(key.clone(), 41);
         pooled.idle();
@@ -298,7 +298,7 @@ mod tests {
     #[test]
     fn test_pool_checkout_returns_none_if_expired() {
         ::futures::lazy(|| {
-            let mut pool = Pool::new(true, Some(Duration::from_secs(1)));
+            let pool = Pool::new(true, Some(Duration::from_secs(1)));
             let key = Rc::new("foo".to_string());
             let mut pooled = pool.pooled(key.clone(), 41);
             pooled.idle();
@@ -310,7 +310,7 @@ mod tests {
 
     #[test]
     fn test_pool_removes_expired() {
-        let mut pool = Pool::new(true, Some(Duration::from_secs(1)));
+        let pool = Pool::new(true, Some(Duration::from_secs(1)));
         let key = Rc::new("foo".to_string());
 
         let mut pooled1 = pool.pooled(key.clone(), 41);
@@ -334,7 +334,7 @@ mod tests {
 
     #[test]
     fn test_pool_checkout_task_unparked() {
-        let mut pool = Pool::new(true, Some(Duration::from_secs(10)));
+        let pool = Pool::new(true, Some(Duration::from_secs(10)));
         let key = Rc::new("foo".to_string());
         let pooled1 = pool.pooled(key.clone(), 41);
 

--- a/src/server/mod.rs
+++ b/src/server/mod.rs
@@ -267,7 +267,7 @@ impl<T> Service for HttpService<T>
     type Error = ::Error;
     type Future = Map<T::Future, fn(Response) -> Message<ResponseHead, http::TokioBody>>;
 
-    fn call(&mut self, message: Self::Request) -> Self::Future {
+    fn call(&self, message: Self::Request) -> Self::Future {
         let (head, body) = match message {
             Message::WithoutBody(head) => (head, http::Body::empty()),
             Message::WithBody(head, body) => (head, body.into()),

--- a/tests/server.rs
+++ b/tests/server.rs
@@ -113,7 +113,7 @@ impl Service for TestService {
     type Response = Response;
     type Error = hyper::Error;
     type Future = Box<Future<Item=Response, Error=hyper::Error>>;
-    fn call(&mut self, req: Request) -> Self::Future {
+    fn call(&self, req: Request) -> Self::Future {
         let tx = self.tx.clone();
         let replies = self.reply.clone();
         req.body().for_each(move |chunk| {

